### ISSUE TITLE
amp-story: Make play/pause button focusable on mobile UI

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32508]
# Timing not adjustable

## WCAG Level
Level A

## Priority
High

## Pages/screens/components affected
* [Navigation](https://www.gstatic.com/amphtml/stamp/qa/navigation.html)
* [Video](https://www.gstatic.com/amphtml/stamp/qa/video.html)
* [Audio - amp-story-page background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-page.html)
* [Audio - amp-story using amp-audio](https://www.gstatic.com/amphtml/stamp/qa/amp-audio.html)
* [Prerender - Video](https://www.gstatic.com/amphtml/stamp/qa/prerender-video.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)

## Description
Content or functionality that relies on a time limit can be problematic for users, as they may not have sufficient time to read/operate a page or to carry out a specific task.

Any process or functionality that happens without being explicitly triggered by the user, after a set time or on a periodic basis, counts as a time limit. This includes partial or full updates of content (for example, a page refresh, being automatically logged-out of an ongoing session), changes to content, or the expiration of a window of opportunity for a user to react to a request for input. It also includes content that is auto-advancing or updating. In other words, animated, moving or scrolling content can also introduce a time limit on a users ability to read, understand, and operate content.

Whenever there is a time limit, users must be able to turn off, adjust, or extend the timeout (unless the timeout is essential, relates to a real-time event, or is longer than 20 hours).

Authors can define auto-advancing for pages in their AMP stories. After a set amount of time, a story advances to the next page.

On large screen/desktop, the "Pause" control in the top-right corner can be used to pause the auto-advancing.

![An AMP story with auto-advancing pages in large-screen/desktop view, with the 'Pause' control highlighted](https://user-images.githubusercontent.com/54933393/107249469-4f35dc80-6a2b-11eb-97f2-28883410cdb2.png)

However, in the small-screen/smartphone sized version, there is no "Pause" control visible. Desktop users that have sized/zoomed their browser window to a small-enough size to trigger the "smartphone" view have no way of pausing the video with their mouse or keyboard. The same problem applies to users on an actual smartphone that are using an external mouse or keyboard.

![The same auto-advancing page in small-screen/smartphone view - the 'Pause' button is not present in this view](https://user-images.githubusercontent.com/54933393/107249521-5bba3500-6a2b-11eb-8f59-398e5f73a18b.png)

Touchscreen users are able to pause the auto-advancing in the small-screen/smartphone view by pressing and holding on the page itself. While arguably this passes the requirement of having a method to "turn off" auto-advancing, this is - subjectively - not a good approach, as it forces users to keep their finger pressed on the touchscreen, just to be able to read/view the content without it disappearing from view.

## User impact
Time limits can affect all users, but users with disabilities (including, but not limited to, users of assistive technologies) are likely to find them particularly problematic. User who are blind, have low vision, have mobility or dexterity impairments, and users with cognitive disabilities may require more time to read content or to perform functions such as filling out on-line forms. Time limits may prevent them from successfully reading content or completing tasks, making a site unusable for them.

## Required solution
Make sure that on both large and small screen view, there is a method for users keyboard, mouse, and touch users to pause any auto-advancing behaviour.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Verify if the page contains any auto-advancing behaviour.
2. Verify that there is a mechanism for users (using mouse, keyboard, touch) to pause the auto-advancing.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.2.1 Timing Adjustable (Level A)](https://www.w3.org/TR/WCAG21/#timing-adjustable)

## Test data
Test date: January 2021

**Comment by gmajoulet** [comment]
Users can long press on the page to pause it. We should treat this as a UX and feature discovery issue more than an accessibility issue, as it impacts all users.

**Comment by patrickhlauke** [comment]
> Users can long press on the page to pause it.

Except for keyboard users (either on desktop that's been sized/zoomed to show the mobile view, or smartphone users that actually need an external bluetooth keyboard to navigate) and assistive technology users (VoiceOver, TalkBack) on smartphones.

**Comment by patrickhlauke** [comment]
suggest making it visible though when it receives focus, as otherwise for sighted keyboard users, their focus will suddenly "disappear" and they won't know why/that they're actually on an invisible play/pause button

**Comment by gmajoulet** [comment]
Great idea thank you. We should also do that for the pagination buttons (go to prev/next page) that are today invisible.

**Comment by processprocess** [comment]
Re-opening due to TalkBack being unable to focus on the button.

**Comment by processprocess** [comment]
@Enriqe This earlier commit you made will allow the button to be focusable and visible with screen reader technology:
https://github.com/ampproject/amphtml/pull/33214/commits/49529f5832cff2e491eaf87abe5990f4d1c6a97a
It looks like it was removed before merging #33214 but I think adding this back will close this.

Please be sure to test it with TalkBack on your end though! 
I'm happy to walk you through steps for setting that up if you want!

**Comment by johocen168** [comment]
Do not understand why the play/pause button should be different on desktop and small-screen/smartphone sized version?
How many user will know long pressing on the page to pause it? Please consider most of common users are not tech savvy.
Amp-story is a great way to interact with user, even better than video, hope to see it getting popular without having to educate  users how to pause. (Also, re-start/re-play icon is important, follow video's icons since every one knows already)
